pkg/konnector/konnector_reconcile.go

```diff
@@ -120,6 +120,16 @@ func (r *reconciler) reconcile(ctx context.Context, binding *kubebindv1alpha1.AP
 	ctrl, err := r.newClusterController(
 		binding.Spec.KubeconfigSecretRef.Namespace+"/"+binding.Spec.KubeconfigSecretRef.Name,
 		providerNamespace,
+		func(svcBinding *kubebindv1alpha1.APIServiceBinding) bool {
+			r.lock.RLock()
+			defer r.lock.RUnlock()
+			// for the very first time, ctrlContext is nil
+			// and need to check for this as well
```

Comment:
don't think this is necessary. The controller starts in line 146 after the `r.controllers` change.